### PR TITLE
Avoid leaking build operation listeners

### PR DESCRIPTION
**Pull request by abstratt**
...when load-after-store=true

### CODE REVIEW

**Comment by abstratt** [comment]
@bot-gradle test ACC

**Comment by abstratt** [comment]
@bot-gradle test ACC

**Comment by abstratt** [comment]
@bot-gradle test ACC

**Comment by abstratt** [comment]
@bot-gradle test ACC

File: subprojects/build-events/src/main/java/org/gradle/internal/build/event/DefaultBuildEventsListenerRegistry.java [hunk 1]
```diff
@@ -117,9 +118,7 @@ public void onTaskCompletion(Provider<? extends OperationCompletionListener> lis
         }
 
         ForwardingBuildEventConsumer subscription = new ForwardingBuildEventConsumer(listenerProvider, executorFactory);
-        if (listenerProvider instanceof RegisteredBuildServiceProvider) {
```
Comment (abstratt) [inline subprojects/build-events/src/main/java/org/gradle/internal/build/event/DefaultBuildEventsListenerRegistry.java:120]: This was being done only for OperationCompletionListeners, but not for BuildOperationListeners.

The most evident symptom was failures around ProgressLoggingFixture, due to the a listener/build service leaking across the load-after-store boundary.

**Comment by abstratt** [comment]
@bot-gradle test ACC

File: subprojects/resources-gcs/build.gradle.kts [hunk 2]
```diff
@@ -36,8 +36,3 @@ dependencies {
 strictCompile {
     ignoreDeprecations()
 }
-
```
Comment (abstratt) [inline subprojects/resources-gcs/build.gradle.kts:39]: The changes here were enough to make the resources-gcs tests pass, as the ProgressLoggingFixture service leakage was the only thing in the way.

**Review by adammurdoch — APPROVED** [review]
This looks like.a good solution to me. Both kinds of listeners are intended to monitor the current invocation of Gradle, regardless of what happens in that invocation. So, preserving the listeners after store, rather than recreating them, makes sense.

**Comment by abstratt** [comment]
@bot-gradle test and merge

**Pull request merged**